### PR TITLE
Update solver statuses for MOI v0.8.

### PR DESCRIPTION
**Pull request by hellemo**
I wasn't sure what to do with the Success statuses, I basically just repeated the alternative status. This might need an overhaul. 

Passing all tests locally on 1.0.3 and 1.1.0rc1 with current master of JuMP, MOI and LQOI.

### CODE REVIEW

File: src/MOIWrapper.jl [hunk 1]
```diff
@@ -364,56 +363,56 @@ function LQOI.get_termination_status(instance::Optimizer)
...
             end
 
         elseif stat_mip == XPR.MIP_Infeasible
             if XPR.hasdualray(instance.inner)
-                return MOI.Success
+                return MOI.INFEASIBLE
             else
-                return MOI.InfeasibleNoResult
+                return MOI.INFEASIBLE
             end
```
Comment (odow) [inline src/MOIWrapper.jl]: No longer any need for the branch. Just return `MOI.INFEASIBLE`

File: src/MOIWrapper.jl [hunk 2]
```diff
@@ -364,56 +363,56 @@ function LQOI.get_termination_status(instance::Optimizer)
...
         elseif stat_mip == XPR.MIP_Optimal
-            return MOI.Success
+            return MOI.OPTIMAL
         elseif stat_mip == XPR.MIP_Unbounded
             if XPR.hasprimalray(instance.inner)
-                return MOI.Success
+                return MOI.INFEASIBLE_OR_UNBOUNDED
             else
-                return MOI.UnboundedNoResult
+                return MOI.INFEASIBLE_OR_UNBOUNDED
```
Comment (odow) [inline src/MOIWrapper.jl]: Did this prove unboundedness? Or dual infeasibility?

Same comment with the branch.

File: src/MOIWrapper.jl [hunk 3]
```diff
@@ -364,56 +363,56 @@ function LQOI.get_termination_status(instance::Optimizer)
...
         elseif stat_lp == XPR.LP_Optimal
-            return MOI.Success
+            return MOI.OPTIMAL
         elseif stat_lp == XPR.LP_Infeasible
             if XPR.hasdualray(instance.inner)
-                return MOI.Success
+                return MOI.INFEASIBLE
             else
-                return MOI.InfeasibleNoResult
+                return MOI.INFEASIBLE
```
Comment (odow) [inline src/MOIWrapper.jl]: Branch again

File: src/MOIWrapper.jl [hunk 4]
```diff
@@ -364,56 +363,56 @@ function LQOI.get_termination_status(instance::Optimizer)
...
+            return MOI.OBJECTIVE_LIMIT
         elseif stat_lp == XPR.LP_Unfinished
             return xprsmoi_stopstatus(instance.inner)
         elseif stat_lp == XPR.LP_Unbounded
             if XPR.hasprimalray(instance.inner)
-                return MOI.Success
+                return MOI.INFEASIBLE_OR_UNBOUNDED
             else
-                return MOI.UnboundedNoResult
+                return MOI.INFEASIBLE_OR_UNBOUNDED
```
Comment (odow) [inline src/MOIWrapper.jl]: Branch again

**Comment by mlubin** [comment]
> what is the difference between INFEASIBLE and INFEASIBLE_OR_UNBOUNDED?

Typically you should report INFEASIBLE_OR_UNBOUNDED only if the solver explicitly says this. I took a quick look at Xpress's `MIPSTATUS` codes and none of them match, so this status should never be returned. I'm surprised that Xpress has only 7 possible values for the MIP status.

File: REQUIRE [hunk 5]
```diff
@@ -1,4 +1,4 @@
 julia 0.6
-MathProgBase 0.6 0.8
-LinQuadOptInterface 0.4 0.5
+MathProgBase 0.6 0.8
```
Comment (blegat) [inline REQUIRE]: Remove trailing whitespace

File: REQUIRE [hunk 6]
```diff
@@ -1,4 +1,4 @@
 julia 0.6
-MathProgBase 0.6 0.8
-LinQuadOptInterface 0.4 0.5
+MathProgBase 0.6 0.8    
+LinQuadOptInterface 0.4
```
Comment (blegat) [inline REQUIRE]: Add a `0.6` upper bound

**Comment by hellemo** [comment]
Thanks for the feedback, everyone, I just did an update with the extra branch removed and the version bounds updated as suggested.

I used INFEASIBLE_OR_UNBOUNDED which seemed to best match the unbounded status from Xpress, but happy for suggestions in both cases.

@joaquimg My 0.6 install isn't cooperating at the moment, will try again tonight.

**Comment by joaquimg** [comment]
The correct match for unbounded status is: DUAL_INFEASIBLE

**Comment by joaquimg** [comment]
We also need to fix the `xprsmoi_stopstatus` function

File: src/MOIWrapper.jl [hunk 7]
```diff
@@ -443,44 +426,44 @@ function LQOI.get_primal_status(instance::Optimizer)
     if XPR.is_mip(instance.inner)
         stat_mip = XPR.get_mip_status2(instance.inner)
         if stat_mip in [XPR.MIP_Solution, XPR.MIP_Optimal]
-            return MOI.FeasiblePoint
+            return MOI.FEASIBLE_POINT
         elseif XPR.MIP_Infeasible && XPR.hasdualray(instance.inner)
-            return MOI.InfeasibilityCertificate
+            return MOI.INFEASIBILITY_CERTIFICATE
```
Comment (joaquimg) [inline src/MOIWrapper.jl]: I think we should have NO_SOLUTION, because the certificate is actually in the dual

Comment (joaquimg) [reply src/MOIWrapper.jl]: even better, just remove this branch

**Comment by hellemo** [comment]
Ah, yes, that makes sense.

Missed the xprsmoi_stopstatus function. What would be the best status for XPR.MIP_LPNotOptimal? OTHER_ERROR?

**Comment by hellemo** [comment]
MOI tests pass on 0.6.4, but the JuMP tests fail. Considering JuMP apparently already dropped 0.6 support, that may not be a big problem. (?)

**Comment by joaquimg** [comment]
What about MPB tests?

> Missed the xprsmoi_stopstatus function. What would be the best status for XPR.MIP_LPNotOptimal? OTHER_ERROR?

lets keep as other error for now.

**Comment by hellemo** [comment]
It seems my fork wasn't quite up to date and that caused my issues in 0.6.4, all tests are passing now as far as I can tell, but there are several warnings.

**Pull request merged**